build.gradle

```diff
@@ -178,6 +179,56 @@ subprojects { subproject ->
 										}
 							}
 
+							if (subproject.plugins.hasPlugin('org.jetbrains.kotlin.jvm')) {
+								def kotlinVersion = '2.2.21'
+								def plugins = asNode().build?.find()?.plugins?.find()
+								if (!plugins) {
+									plugins = asNode().appendNode('build').with {
+										appendNode('sourceDirectory', '${project.basedir}/src/main/kotlin')
+										appendNode('testSourceDirectory', '${project.basedir}/src/test/kotlin')
+										appendNode('plugins')
+									}
+								}
+
+								plugins.appendNode('plugin')
+										.with {
+											appendNode('groupId', 'org.jetbrains.kotlin')
+											appendNode('artifactId', 'kotlin-maven-plugin')
+											appendNode('version', kotlinVersion)
+											appendNode('configuration').with {
+												appendNode('args').with {
+													appendNode('arg', '-Xjsr305=strict')
+												}
+												appendNode('compilerPlugins').with {
+													appendNode('plugin', 'spring')
+												}
+											}
+											appendNode('executions').with {
+												appendNode('execution').with {
+													appendNode('id', 'compile')
+													appendNode('phase', 'compile')
+													appendNode('goals').with {
+														appendNode('goal', 'compile')
+													}
+												}
+												appendNode('execution').with {
```

Comment:
Do we really need this `execution` to be expressed for Maven?
Looks like start.spring.io does not suggest that one.

Reply:
If it is not included then no classes are generated at compile time.